pkg/diag/validator/pod.go

```diff
@@ -209,34 +209,35 @@ func processPodEvents(e corev1.EventInterface, pod v1.Pod, ps *podStatus) {
 	}
 	switch recentEvent.Reason {
 	case failedScheduling:
-		ps.statusCode = proto.StatusCode_STATUSCHECK_FAILED_SCHEDULING
-		ps.err = fmt.Errorf(recentEvent.Message)
+		ps.ae.ErrCode = proto.StatusCode_STATUSCHECK_FAILED_SCHEDULING
+		ps.ae.Message = recentEvent.Message
 	case unhealthy:
-		ps.statusCode = proto.StatusCode_STATUSCHECK_UNHEALTHY
-		ps.err = fmt.Errorf(recentEvent.Message)
+		ps.ae.ErrCode = proto.StatusCode_STATUSCHECK_UNHEALTHY
+		ps.ae.Message = recentEvent.Message
 	default:
 		// TODO: Add unique error codes for reasons
-		ps.statusCode = proto.StatusCode_STATUSCHECK_UNKNOWN_EVENT
-		ps.err = fmt.Errorf("%s: %s", recentEvent.Reason, recentEvent.Message)
+		ps.ae.ErrCode = proto.StatusCode_STATUSCHECK_UNKNOWN_EVENT
+		ps.ae.Message = fmt.Sprintf("%s: %s", recentEvent.Reason, recentEvent.Message)
 	}
 }
 
 type podStatus struct {
-	name       string
-	namespace  string
-	phase      string
-	logs       []string
-	err        error
-	statusCode proto.StatusCode
+	name      string
+	namespace string
+	phase     string
+	logs      []string
+	ae        *proto.ActionableErr
 }
 
 func (p *podStatus) isStable() bool {
-	return p.phase == success || (p.phase == running && p.err == nil)
+	return p.phase == success || (p.phase == running && p.ae.Message == "")
```

Comment:
Shouldn't this code check that `p.ae != nil`?

Reply:
with this changed to a struct, not needed anymore